### ISSUE TITLE
Enhancement: option in context menu to allow message translation.

### ISSUE THREAD

**Issue by ianmacd** [oxen-io/session-desktop#2234]
It would be a huge convenience to be able to right-click a message bubble and invoke translation of that message.

**Comment by Bilb** [comment]
Considering how google can have poor translations on some languages, and that we would need an offline solution I don't think this happening

**Comment by ianmacd** [comment]
Google isn't the only game in town, but its translations for a large number of languages are excellent. It's really progressed over the years.

Isn't it throwing the baby out with the bathwater to reject the feature on the basis of translations into a lesser used language being less than perfect? After all, this is machine learning. It's never perfect and is always presented as a best effort feature.

Have a look at the way the unofficial Telegram Unigram client implements this on Windows. It's a very nice interface. The **official** desktop client for Telegram doesn't have the feature, but the mobile client does.

I don't see why you'd need an off-line solution to be able to implement the future at all. Make it on-line only.

**Comment by KeeJef** [comment]
I really don't think this is a good idea if the solution isn't offline, because I really don't want to be leaking text content of actual messages to any third party. Even if its done at the users request, and even if that connection is onion routed. Its just too easy for people who don't know better to leak an entire log

**Comment by ianmacd** [comment]
> I really don't think this is a good idea if the solution isn't offline, because I really don't want to be leaking text content of actual messages to any third party. Even if its done at the users request, and even if that connection is onion routed. Its just too easy for people who don't know better to leak an entire log

I respectfully disagree with the decision, because forgoing functionality for the sake of "people who don't know better" is a fatally flawed pursuit. As Douglas Adams once wrote, "A common mistake that people make when trying to design something completely foolproof is to underestimate the ingenuity of complete fools."

Regardless, I don't see how a user could leak an entire log if message translation were on a per message basis. Such a user would need to individually select each message in an entire conversation and submit each of them in turn for translation. Even then, this would leak only one side of the conversation.

Here's a possible compromise: What if a translation option were only available for the messages of open groups? Since these are public anyway, there can be no reasonable claim of even accidental "leaking".

Would you be open to that?

**Comment by KeeJef** [comment]
Yeah i could see this being useful in open groups, ill put this into the backlog

**Comment by JEDUMAR** [comment]
Sin dudar me gustaría que SESSION tuviese la opción de traducción siempre y cuando sea dentro de la seguridad y privacidad, sin poner en riesgo la naturaleza de session... Tambien me gustaria demasiado poder incluir una actividad también privada dentro des condiciones  que haga funciones de agenda y recordatorios poder ordenar nuestras actividades diarias como agenda pero con privacidad y seguridad dentro de la plataforma. Ayudaría mucho a tener una organización la idea es darle mayor usabilidad y no limitar esta maravillosa aplicación que restituye nuestros Derechos civiles y fundamentales.
